runners/core-java/src/main/java/org/apache/beam/runners/core/SystemReduceFn.java

```diff
@@ -40,15 +40,18 @@
 public abstract class SystemReduceFn<K, InputT, AccumT, OutputT, W extends BoundedWindow>
     extends ReduceFn<K, InputT, OutputT, W> {
   private static final String BUFFER_NAME = "buf";
+  private static final String BUFFER_RETRACTION = "buf_retraction";
```

Comment:
Note that we need a smarter implementation on SystemReduceFn to achieve a "incremental" style retracting. E.g. retractions do not trigger a re-computation but just apply delta.

Reply:
Even my current implementation is neither sufficient or correct (it's just a PoC).